src/megaapi_impl.cpp

```diff
@@ -9290,27 +9299,30 @@ void MegaApiImpl::fireOnStreamingStart(MegaTransferPrivate *transfer)
         (*it)->onTransferStart(api, transfer);
 }
 
-void MegaApiImpl::fireOnStreamingTemporaryError(MegaTransferPrivate *transfer, MegaError e)
+void MegaApiImpl::fireOnStreamingTemporaryError(MegaTransferPrivate *transfer, MegaErrorPrivate *e)
 {
     for(set<MegaTransferListener *>::iterator it = httpServerListeners.begin(); it != httpServerListeners.end() ; it++)
-        (*it)->onTransferTemporaryError(api, transfer, &e);
+        (*it)->onTransferTemporaryError(api, transfer, e);
+
+    delete e;
```

Comment:
as noted above, the parameter here can be `std::unique_ptr<MegaErrorPrivate> e`
and then ownership is tracked, human readers and compilers know what is going on, and we don't need the manual call to `delete`    (the onTransferTemporaryError would use `e.get()` )

Reply:
Agree

Reply:
It's solved at Matt PR #2045